### PR TITLE
docs: add initial how-to doc

### LINKED ISSUE #24: How-to Documentation
# This is a Feature Proposal

## Description

Add a how-to doc that will help end users navigate the different commands based on what they want to accomplish.

- What is the use case that should be solved. The more detail you describe this in the easier it is to understand for us.
- If there is additional config how would it look

Similar or dependent issues:

- #12345

### PR DESCRIPTION
**Pull request by elizabethhalper**
## What did you implement:

Closes #24 

This is the initial commit for the how-to documentation. Written with end users in mind to guide them through commands and variables they will need to accomplish specific tasks using `pjr`. Also written with the intention of fleshing out as features get added. Please correct any incorrect assumptions I've made!

## How did you implement it:

Wrote it down :)

## Todos:

_**Note: Run `npm run check` to run all validation checks on proposed changes**_

- [X] Ensure there are no lint errors
       **Validate via `npm run lint`**  
       _Note: Some reported issues can be automatically fixed by running `npm run lint:fix`_
- [X] Write tests and confirm existing functionality is not broken
       **Validate via `npm test`**
- [X] Write documentation
- [X] Pull request title and commit message follows [the conventional commit specification](https://www.conventionalcommits.org/en/v1.0.0-beta.2/#summary)
- [X] Provide verification config / commands / resources
- [ ] Requested reviews from at least 2 individuals
- [X] Assigned PR to myself
- [X] Added PR to correct project
- [X] This is ready for review

**_Is it a breaking change?:_** NO

### CODE REVIEW

File: docs/how_to.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+### Azure DevOps
+
+- Agile provider (currently only supports and defaults to 'azdo') (default: "azdo")
+- Base URL for your DevOps organization
+  - This usually has the format '<https://dev.azure.com/{organization}>'
+  - If you are unsure how to create an organization, follow [these guidelines](https://docs.microsoft.com/en-us/azure/devops/organizations/accounts/create-organization?view=azure-devops).
+- Desired name for the agile provider project
+- DevOps access token
+  - Go to 'https://dev.azure.com/{organization}/_usersSettings/tokens' to generate a token
```
Comment (tbarlow12) [inline docs/how_to.md:15]: I think we can probably organize this section a little more. Maybe we can show what an example `agile` config would look like for Azure DevOps and explain the various parts. Something like...

#### What you need
- **baseUrl** - The base URL for your Azure DevOps organization. This usually has the format `https://dev.azure.com/{organizationName}`. If you are unsure how to create an organization, follow [these guidelines](https://docs.microsoft.com/en-us/azure/devops/organizations/accounts/create-organization?view=azure-devops).
- **accessToken** - A personal access token for your AzureDevOps organization. Go to `https://dev.azure.com/{organization}/_usersSettings/tokens` to generate a token.
- **projectName** - Name of your Azure DevOps project. `projector` does not currently support project creation, but will soon. Currently must be an existing project.

#### Example Config

```
...
"agile": {
    "providerName": "azdo",
    "providerOptions": {
        "baseUrl": "{baseUrl}",
        "projectName": "{projectName}",
        "personalAccessToken": "{accessToken}"
    }
    ...
}
```

**Review by tbarlow12 — APPROVED** [review]
This is great. We can continue to iterate this on a team as well. I'd like to include a link to the How To doc in the README somewhere and I had a few other things to address before merging

File: docs/commands.md [hunk 2]
```diff
@@ -22,7 +22,7 @@
     - [`pjr playbook template`](#pjr-playbook-template) - Discover and use templates from the code-with engineering playbook
       - [`pjr playbook template copy`](#pjr-playbook-template-copy) - Copy templates from playbook to local working directory
   - [`pjr project`](#pjr-project) - Local Projector configuration
-    - [`pjr project init`](#pjr-project-init) - Initialize local projector configuration
+    - [`pjr project init`](#pjr-project-init) - Initialize local projector configuration in `projector.json` file
```
Comment (zmmille2) [inline docs/commands.md:25]: This is an auto-generated file. If you wanna update stuff here (which I think is great) update the source 😁

File: package.json [hunk 3]
```diff
@@ -32,7 +32,7 @@
   "license": "ISC",
   "dependencies": {
     "@octokit/rest": "18.0.12",
-    "axios": "0.21.0",
+    "axios": "^0.21.1",
```
Comment (zmmille2) [inline package.json:35]: please pin!

File: README.md [hunk 4]
```diff
@@ -2,6 +2,13 @@
...
 
+----------
+
+Projector is a tool that reduces repetitive work by creating CSE projects and backlogs using CLI commands. By running just a few commands, the tool sets up a project and backlog structure that reinforces good engineering fundamentals, but is also flexible enough for customers by allowing building blocks of a project to be mixed and matched.
+
+## Table of Contents
+
+- [Table of Contents](#table-of-contents)
```
Comment (zmmille2) [inline README.md:11]: Might be good to link to the `how_to.md` from the README.md somewhere.

File: README.md [hunk 5]
```diff
@@ -2,6 +2,13 @@
...
 
+----------
+
+Projector is a tool that reduces repetitive work by creating CSE projects and backlogs using CLI commands. By running just a few commands, the tool sets up a project and backlog structure that reinforces good engineering fundamentals, but is also flexible enough for customers by allowing building blocks of a project to be mixed and matched.
```
Comment (zmmille2) [inline README.md:7]: nit: I prefer sentences on newlines in Markdown.

File: docs/how_to.md [hunk 6]
```diff
@@ -0,0 +1,53 @@
+# How To
+
+## Initialize projector
+
+In order to initialize your local machine to use projector, you will run [`pjr project init`](./commands.md#pjr-project-init), but first you need the values shown below depending on the provider you choose to use. This command puts your values into a `projector.json` file that will be used as a configuration file when .
```
Comment (zmmille2) [inline docs/how_to.md:5]: This paragraph needs a little love.
1. The sentence ends with "when ." When what?
2. We should provide the commands in the order people need them. "but first" is a bit of a red flag here - maybe we should rephrase?

**Review by zmmille2 — CHANGES_REQUESTED** [review]

File: package.json (hunk 3 above)
Comment (zmmille2) [reply package.json:35]: Tanner covered this - resolving

File: docs/how_to.md [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+
+### Azure DevOps
+
+#### Creating Sprints
+
+In order to create sprints, the user should run [`pjr agile sprints create`](./commands.md#pjr-agile-sprints-create). Creating work items is in progress, using templates is also [in the backlog](https://github.com/projector-cli/projector/issues/6).
+
+### JIRA
+
+Currently not supported, but is [in the backlog](https://github.com/projector-cli/projector/issues/14).
```
[the remark this answers is not in the document]
Comment (elizabethhalper) [reply docs/how_to.md:49]: Fixed!

File: docs/how_to.md (hunk 6 above)
Comment (elizabethhalper) [reply docs/how_to.md:5]: Good call! Revising now

File: docs/how_to.md (hunk 1 above)
Comment (elizabethhalper) [reply docs/how_to.md:15]: Great suggestions! Fixed!

File: docs/commands.md (hunk 2 above)
Comment (elizabethhalper) [reply docs/commands.md:25]: I believe I did only update it there, but I also ran `npm run docs:generate` and it showed up in diffs

File: README.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (elizabethhalper) [reply README.md:7]: Yep its definitely a duplicate. I had written it before you wrote yours so good thing they look pretty much the same ;) I'll delete this and we can keep the `About` section!

File: README.md (hunk 4 above)
Comment (elizabethhalper) [reply README.md:11]: Added it to the "Getting Started" section

File: README.md [hunk 8]
```diff
@@ -2,6 +2,13 @@
...
+
+## Table of Contents
```
[the remark this answers is not in the document]
Comment (alancast) [reply README.md:5]: I think it needs to be on the same line as the header, it still seems to be showing up in the TOC

File: package.json [hunk 9]
```diff
@@ -32,7 +32,7 @@
   "license": "ISC",
   "dependencies": {
     "@octokit/rest": "18.0.12",
-    "axios": "0.21.0",
+    "axios": "0.21.1",
```
Comment (alancast) [inline package.json:35]: I think this change has already been merged into master :-) So I think this and the package-lock.json can probably get reverted

File: docs/how_to.md [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+### Example Config
+
+```
+{
+    "agile": {
+        "providerName": "azdo",
+        "providerOptions": {
+            "baseUrl": "https://dev.azure.com/my-org-name/"
+            "projectName": "my-test-project",
+            "personalAccessToken": "azd0spa6tl08ksl1kea6unch0fn3mbe8san6l3tt3rs"
```
Comment (alancast) [inline docs/how_to.md:23]: lolol this is incredible

File: docs/how_to.md [hunk 11]
```diff
@@ -0,0 +1,72 @@
...
+
+## Create a code repo
+
+### Azure DevOps
+
+Currently not supported, but is [in the backlog](https://github.com/projector-cli/projector/issues/20).
+
+### Github
+
+Currently not supported, but is [in the backlog](https://github.com/projector-cli/projector/issues/20).
```
Comment (alancast) [inline docs/how_to.md:54]: nit: I feel like creating a repo in Azure Devops and in GitHub should likely be separate issues. They feel like they could have a lot of different things.

I know it's not part of this PR, but perhaps splitting those out into two separate issues if you agree

Comment (alancast) [reply docs/how_to.md:54]: But perhaps I'm wrong and they are basically the same. Some company named Microsoft does own both of them

**Review by alancast — APPROVED** [review]
Left a couple of suggestions but a super helpful doc and good to have around!

File: README.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (elizabethhalper) [reply README.md:5]: Fixed!

File: docs/how_to.md (hunk 11 above)
Comment (elizabethhalper) [reply docs/how_to.md:54]: Hahah I totally agree, it'll be two separate issues. The issue itself acknowledges that and also says that it'll probably be split into multiple issues once implementation starts. It also mentions adding templated README's and working agreements so there are a lot of stories that need to come from it and I think I'd rather update it when we have more of it fleshed out.

File: package.json (hunk 9 above)
Comment (elizabethhalper) [reply package.json:35]: Good catch! I'll just rebase and hopefully that will fix it

**Review by zmmille2 — APPROVED** [review]

**Pull request merged**